src/lib.rs

```diff
@@ -973,3 +980,29 @@ fn getenv_unwrap(v: &str) -> String {
 fn fail(s: &str) -> ! {
     panic!("\n{}\n\nbuild script failed, must exit now", s)
 }
+
+// Returns the major, minor and patch versions of cmake
+fn cmake_version(executable: &OsStr) -> Result<(u8, u8, u8), Box<dyn std::error::Error>> {
+    let cmd = Command::new(executable).arg("--version").output()?;
+    let version = String::from_utf8_lossy(&cmd.stdout);
+    let version: Vec<&str> = version.split_whitespace().collect();
+    let version: Vec<&str> = version[2].split('.').collect();
+    let mut temp = vec![];
+    for i in version.iter() {
+        let curr = i.parse()?;
+        temp.push(curr);
+    }
+    Ok((temp[0], temp[1], temp[2]))
+}
+
+fn cmake_has_parallel(executable: &OsStr) -> bool {
+    if let Ok(cmake_version) = cmake_version(&executable) {
+        if (cmake_version.0 > 3) || (cmake_version.0 == 3 && cmake_version.1 >= 12) {
```

Comment:
I think this would work as `if cmake_version >= (3, 12, 0)`

Reply:
huh it does work